src/Akeneo/Apps/front/src/application/apps/components/CreateAppForm.tsx

```diff
@@ -0,0 +1,176 @@
+import React, {ChangeEvent, useRef, useEffect, useContext, useReducer} from 'react';
+import {FlowType} from '../../../domain/apps/flow-type.enum';
+import {ApplyButton, Form, FormGroup, FormInput} from '../../common';
+import {FlowTypeHelper} from './FlowTypeHelper';
+import {FlowTypeSelect} from './FlowTypeSelect';
+import {appFormReducer, CreateFormState} from '../reducers/app-form-reducer';
+import {inputChanged, setError, formIsInvalid, formIsValid} from '../actions/create-form-actions';
+import {sanitize} from '../../shared/sanitize';
+import {Translate, TranslateContext} from '../../shared/translate';
+import {fetch} from '../../shared/fetch';
+import {isErr} from '../../shared/fetch/result';
+import {NotificationLevel, useNotify} from '../../shared/notify';
+import {useRoute} from '../../shared/router';
+import {useHistory} from 'react-router';
+
+const initialState: CreateFormState = {
+    controls: {
+        code: {name: 'code', value: '', errors: {}, dirty: false, valid: false},
+        label: {name: 'label', value: '', errors: {}, dirty: false, valid: false},
+        flow_type: {
+            name: 'flow_type',
+            value: FlowType.DATA_SOURCE,
+            errors: {},
+            dirty: false,
+            valid: true,
+        },
+    },
+    valid: false,
+};
+
+interface ResultError {
+    message: string;
+    errors: Array<{
+        name: string;
+        reason: string;
+    }>;
+}
+
+export const CreateAppForm = () => {
+    const [state, dispatch] = useReducer(appFormReducer, initialState);
+
+    const codeInputRef = useRef<HTMLInputElement>(null);
+    const labelInputRef = useRef<HTMLInputElement>(null);
+
+    const url = useRoute('akeneo_apps_create_rest');
+    const notify = useNotify();
+    const translate = useContext(TranslateContext);
+    const history = useHistory();
+    const isFirstRender = useRef(true);
+
+    useEffect(() => {
+        if (isFirstRender.current) {
+            isFirstRender.current = false;
+
+            return;
+        }
+        [codeInputRef, labelInputRef].forEach(inputRef => {
+            const input = inputRef.current;
+            if (null === input) {
+                return;
+            }
+
+            const name = input.name;
+            if (false === input.checkValidity()) {
```

Comment:
You might be able to do this check to avoid re-validating. If you have errors on an invalid input, then it's already validated.
```suggestion
            if (false === input.checkValidity() && 0 === Object.keys(state.controls[name].errors).length) {
```